src/runner_metrics.py

```diff
@@ -139,32 +175,20 @@ def _extract_metrics_from_fs(fs: shared_fs.SharedFilesystem) -> Optional[RunnerM
         return None
 
     try:
-        pre_job_metrics = json.loads(fs.path.joinpath(PRE_JOB_METRICS_FILE_NAME).read_text())
-        logger.debug("Pre-job metrics for runner %s: %s", runner_name, pre_job_metrics)
-    except FileNotFoundError:
-        logger.warning("%s not found for runner %s.", PRE_JOB_METRICS_FILE_NAME, runner_name)
-        return None
-    except JSONDecodeError as exc:
-        raise CorruptMetricDataError(str(exc)) from exc
-
-    try:
-        post_job_metrics = json.loads(fs.path.joinpath(POST_JOB_METRICS_FILE_NAME).read_text())
-        logger.debug("Post-job metrics for runner %s: %s", runner_name, post_job_metrics)
-    except FileNotFoundError:
-        logger.warning("%s not found for runner %s", POST_JOB_METRICS_FILE_NAME, runner_name)
-        post_job_metrics = None
-    except JSONDecodeError as exc:
-        raise CorruptMetricDataError(str(exc)) from exc
-
-    if not isinstance(pre_job_metrics, dict):
-        raise CorruptMetricDataError(
-            f"Pre-job metrics for runner {runner_name} is not a JSON object."
+        pre_job_metrics = _extract_metrics_from_fs_file(
+            fs=fs, runner_name=runner_name, filename=PRE_JOB_METRICS_FILE_NAME
         )
+        if not pre_job_metrics:
+            return None
+        logger.debug("Pre-job metrics for runner %s: %s", runner_name, pre_job_metrics)
 
-    if not isinstance(post_job_metrics, dict) and post_job_metrics is not None:
-        raise CorruptMetricDataError(
-            f"Post-job metrics for runner {runner_name} is not a JSON object."
+        post_job_metrics = _extract_metrics_from_fs_file(
+            fs=fs, runner_name=runner_name, filename=POST_JOB_METRICS_FILE_NAME
         )
+        logger.debug("Post-job metrics for runner %s: %s", runner_name, post_job_metrics)
+    except CorruptMetricDataError as exc:
+        logger.error("Error extracting metrics from shared filesystem, %s", exc)
+        raise
```

Comment:
I think the try except and raise is not necessary here (the fct later raises CorruptMetricDataError).

Reply:
I think it might be better to separate the two `CorruptMetricDataError`, one that can be raised from FS interaction
the other validation error from pydantic.
The reason I think it should be separated is because it is better to be explicit about what functions could be responsible for the error, rather than letting the error bleed/bubble up. For example, when we decide to change the error thrown by the `_extract_metrics_from_fs_file` as `ExtractMetricsError` from `CorruptMetricDataError`, it is easier to notice the change since it is wrapped in try/except. If we let it bubble up without the try/except, we might not have caught it and cause an error which is harder to catch later on.

Would this be a reasonable opinion to keep this?

Reply:
Yes, I understand the reasoning. But could you delete the error log in line 190? The fct `_extract_fs` will log the whole exception trace in line 244.